// devextreme-angular/__ivy_ngcc__/index.js
var ɵngcc0 = require('./ui/nested/validation-rule-dxi');
var ɵngcc1 = require('./ui/nested/tab-dxi');
var ɵngcc2 = require('./ui/nested/location-dxi');
var ɵngcc3 = require('./ui/nested/item-dxi');
var ɵngcc4 = require('./ui/nested/button-dxi');
var ɵngcc5 = require('./ui/nested/options');
var ɵngcc6 = require('./ui/nested/animation');
var ɵngcc7 = require('./ui/nested/export');
var ɵngcc8 = require('./ui/nested/geometry');
var ɵngcc9 = require('./ui/nested/label');
var ɵngcc10 = require('./ui/nested/font');
var ɵngcc11 = require('./ui/nested/format');
var ɵngcc12 = require('./ui/nested/legend');
var ɵngcc13 = require('./ui/nested/border');
var ɵngcc14 = require('./ui/nested/item-text-format');
var ɵngcc15 = require('./ui/nested/margin');
var ɵngcc16 = require('./ui/nested/title');
var ɵngcc17 = require('./ui/nested/subtitle');
var ɵngcc18 = require('./ui/nested/loading-indicator');
var ɵngcc19 = require('./ui/nested/size');
var ɵngcc20 = require('./ui/nested/tooltip');
var ɵngcc21 = require('./ui/nested/shadow');
var ɵngcc22 = require('./ui/nested/box');
var ɵngcc23 = require('./ui/nested/adaptive-layout');
var ɵngcc24 = require('./ui/nested/annotation-dxi');
var ɵngcc25 = require('./ui/nested/image');
var ɵngcc26 = require('./ui/nested/break-dxi');
var ɵngcc27 = require('./ui/nested/constant-line-dxi');
var ɵngcc28 = require('./ui/nested/strip-dxi');
var ɵngcc29 = require('./ui/nested/argument-axis');
var ɵngcc30 = require('./ui/nested/aggregation-interval');
var ɵngcc31 = require('./ui/nested/break-style');
var ɵngcc32 = require('./ui/nested/constant-line-style');
var ɵngcc33 = require('./ui/nested/grid');
var ɵngcc34 = require('./ui/nested/minor-grid');
var ɵngcc35 = require('./ui/nested/minor-tick');
var ɵngcc36 = require('./ui/nested/minor-tick-interval');
var ɵngcc37 = require('./ui/nested/min-visual-range-length');
var ɵngcc38 = require('./ui/nested/strip-style');
var ɵngcc39 = require('./ui/nested/tick');
var ɵngcc40 = require('./ui/nested/tick-interval');
var ɵngcc41 = require('./ui/nested/common-annotation-settings');
var ɵngcc42 = require('./ui/nested/common-axis-settings');
var ɵngcc43 = require('./ui/nested/common-pane-settings');
var ɵngcc44 = require('./ui/nested/common-series-settings');
var ɵngcc45 = require('./ui/nested/aggregation');
var ɵngcc46 = require('./ui/nested/area');
var ɵngcc47 = require('./ui/nested/hover-style');
var ɵngcc48 = require('./ui/nested/hatching');
var ɵngcc49 = require('./ui/nested/connector');
var ɵngcc50 = require('./ui/nested/point');
var ɵngcc51 = require('./ui/nested/height');
var ɵngcc52 = require('./ui/nested/url');
var ɵngcc53 = require('./ui/nested/width');
var ɵngcc54 = require('./ui/nested/selection-style');
var ɵngcc55 = require('./ui/nested/reduction');
var ɵngcc56 = require('./ui/nested/value-error-bar');
var ɵngcc57 = require('./ui/nested/bar');
var ɵngcc58 = require('./ui/nested/bubble');
var ɵngcc59 = require('./ui/nested/candlestick');
var ɵngcc60 = require('./ui/nested/fullstackedarea');
var ɵngcc61 = require('./ui/nested/fullstackedbar');
var ɵngcc62 = require('./ui/nested/fullstackedline');
var ɵngcc63 = require('./ui/nested/fullstackedspline');
var ɵngcc64 = require('./ui/nested/fullstackedsplinearea');
var ɵngcc65 = require('./ui/nested/argument-format');
var ɵngcc66 = require('./ui/nested/line');
var ɵngcc67 = require('./ui/nested/rangearea');
var ɵngcc68 = require('./ui/nested/rangebar');
var ɵngcc69 = require('./ui/nested/scatter');
var ɵngcc70 = require('./ui/nested/spline');
var ɵngcc71 = require('./ui/nested/splinearea');
var ɵngcc72 = require('./ui/nested/stackedarea');
var ɵngcc73 = require('./ui/nested/stackedbar');
var ɵngcc74 = require('./ui/nested/stackedline');
var ɵngcc75 = require('./ui/nested/stackedspline');
var ɵngcc76 = require('./ui/nested/stackedsplinearea');
var ɵngcc77 = require('./ui/nested/steparea');
var ɵngcc78 = require('./ui/nested/stepline');
var ɵngcc79 = require('./ui/nested/stock');
var ɵngcc80 = require('./ui/nested/crosshair');
var ɵngcc81 = require('./ui/nested/horizontal-line');
var ɵngcc82 = require('./ui/nested/vertical-line');
var ɵngcc83 = require('./ui/nested/data-prepare-settings');
var ɵngcc84 = require('./ui/nested/pane-dxi');
var ɵngcc85 = require('./ui/nested/scroll-bar');
var ɵngcc86 = require('./ui/nested/series-dxi');
var ɵngcc87 = require('./ui/nested/series-template');
var ɵngcc88 = require('./ui/nested/value-axis-dxi');
var ɵngcc89 = require('./ui/nested/zoom-and-pan');
var ɵngcc90 = require('./ui/nested/drag-box-style');
var ɵngcc91 = require('./ui/nested/range-dxi');
var ɵngcc92 = require('./ui/nested/range-container');
var ɵngcc93 = require('./ui/nested/scale');
var ɵngcc94 = require('./ui/nested/subvalue-indicator');
var ɵngcc95 = require('./ui/nested/text');
var ɵngcc96 = require('./ui/nested/value-indicator');
var ɵngcc97 = require('./ui/nested/hide');
var ɵngcc98 = require('./ui/nested/show');
var ɵngcc99 = require('./ui/nested/position');
var ɵngcc100 = require('./ui/nested/at');
var ɵngcc101 = require('./ui/nested/boundary-offset');
var ɵngcc102 = require('./ui/nested/collision');
var ɵngcc103 = require('./ui/nested/my');
var ɵngcc104 = require('./ui/nested/offset');
var ɵngcc105 = require('./ui/nested/show-event');
var ɵngcc106 = require('./ui/nested/show-submenu-mode');
var ɵngcc107 = require('./ui/nested/delay');
var ɵngcc108 = require('./ui/nested/column-chooser');
var ɵngcc109 = require('./ui/nested/column-fixing');
var ɵngcc110 = require('./ui/nested/texts');
var ɵngcc111 = require('./ui/nested/column-dxi');
var ɵngcc112 = require('./ui/nested/header-filter');
var ɵngcc113 = require('./ui/nested/lookup');
var ɵngcc114 = require('./ui/nested/form-item');
var ɵngcc115 = require('./ui/nested/editing');
var ɵngcc116 = require('./ui/nested/form');
var ɵngcc117 = require('./ui/nested/col-count-by-screen');
var ɵngcc118 = require('./ui/nested/tab-panel-options');
var ɵngcc119 = require('./ui/nested/button-options');
var ɵngcc120 = require('./ui/nested/toolbar-item-dxi');
var ɵngcc121 = require('./ui/nested/popup');
var ɵngcc122 = require('./ui/nested/custom-operation-dxi');
var ɵngcc123 = require('./ui/nested/field-dxi');
var ɵngcc124 = require('./ui/nested/filter-builder');
var ɵngcc125 = require('./ui/nested/filter-operation-descriptions');
var ɵngcc126 = require('./ui/nested/group-operation-descriptions');
var ɵngcc127 = require('./ui/nested/filter-builder-popup');
var ɵngcc128 = require('./ui/nested/filter-panel');
var ɵngcc129 = require('./ui/nested/filter-row');
var ɵngcc130 = require('./ui/nested/operation-descriptions');
var ɵngcc131 = require('./ui/nested/grouping');
var ɵngcc132 = require('./ui/nested/group-panel');
var ɵngcc133 = require('./ui/nested/keyboard-navigation');
var ɵngcc134 = require('./ui/nested/load-panel');
var ɵngcc135 = require('./ui/nested/master-detail');
var ɵngcc136 = require('./ui/nested/pager');
var ɵngcc137 = require('./ui/nested/paging');
var ɵngcc138 = require('./ui/nested/remote-operations');
var ɵngcc139 = require('./ui/nested/scrolling');
var ɵngcc140 = require('./ui/nested/search-panel');
var ɵngcc141 = require('./ui/nested/selection');
var ɵngcc142 = require('./ui/nested/sort-by-group-summary-info-dxi');
var ɵngcc143 = require('./ui/nested/sorting');
var ɵngcc144 = require('./ui/nested/state-storing');
var ɵngcc145 = require('./ui/nested/group-item-dxi');
var ɵngcc146 = require('./ui/nested/total-item-dxi');
var ɵngcc147 = require('./ui/nested/summary');
var ɵngcc148 = require('./ui/nested/value-format');
var ɵngcc149 = require('./ui/nested/calendar-options');
var ɵngcc150 = require('./ui/nested/display-format');
var ɵngcc151 = require('./ui/nested/drop-down-options');
var ɵngcc152 = require('./ui/nested/item');
var ɵngcc153 = require('./ui/nested/media-resizing');
var ɵngcc154 = require('./ui/nested/mention-dxi');
var ɵngcc155 = require('./ui/nested/toolbar');
var ɵngcc156 = require('./ui/nested/variables');
var ɵngcc157 = require('./ui/nested/menu-item-dxi');
var ɵngcc158 = require('./ui/nested/search-editor-options');
var ɵngcc159 = require('./ui/nested/center-dxi');
var ɵngcc160 = require('./ui/nested/key');
var ɵngcc161 = require('./ui/nested/marker-dxi');
var ɵngcc162 = require('./ui/nested/route-dxi');
var ɵngcc163 = require('./ui/nested/show-first-submenu-mode');
var ɵngcc164 = require('./ui/nested/small-values-grouping');
var ɵngcc165 = require('./ui/nested/data-source');
var ɵngcc166 = require('./ui/nested/store');
var ɵngcc167 = require('./ui/nested/field-chooser');
var ɵngcc168 = require('./ui/nested/field-panel');
var ɵngcc169 = require('./ui/nested/value-axis');
var ɵngcc170 = require('./ui/nested/hide-event');
var ɵngcc171 = require('./ui/nested/background');
var ɵngcc172 = require('./ui/nested/behavior');
var ɵngcc173 = require('./ui/nested/chart');
var ɵngcc174 = require('./ui/nested/indent');
var ɵngcc175 = require('./ui/nested/marker');
var ɵngcc176 = require('./ui/nested/max-range');
var ɵngcc177 = require('./ui/nested/min-range');
var ɵngcc178 = require('./ui/nested/shutter');
var ɵngcc179 = require('./ui/nested/slider-handle');
var ɵngcc180 = require('./ui/nested/slider-marker');
var ɵngcc181 = require('./ui/nested/col-dxi');
var ɵngcc182 = require('./ui/nested/row-dxi');
var ɵngcc183 = require('./ui/nested/link');
var ɵngcc184 = require('./ui/nested/node');
var ɵngcc185 = require('./ui/nested/resource-dxi');
var ɵngcc186 = require('./ui/nested/view-dxi');
var ɵngcc187 = require('./ui/nested/colorizer');
var ɵngcc188 = require('./ui/nested/group');
var ɵngcc189 = require('./ui/nested/tile');
var ɵngcc190 = require('./ui/nested/adapter');
var ɵngcc191 = require('./ui/nested/control-bar');
var ɵngcc192 = require('./ui/nested/layer-dxi');
var ɵngcc193 = require('./ui/nested/legend-dxi');
var ɵngcc194 = require('./ui/nested/source');
"use strict";
function __export(m) {
    for (var p in m) if (!exports.hasOwnProperty(p)) exports[p] = m[p];
}
/*!
 * devextreme-angular
 * Version: 19.1.11
 * Build date: Fri May 15 2020
 *
 * Copyright (c) 2012 - 2020 Developer Express Inc. ALL RIGHTS RESERVED
 *
 * This software may be modified and distributed under the terms
 * of the MIT license. See the LICENSE file in the root of the project for details.
 *
 * https://github.com/DevExpress/devextreme-angular
 */
Object.defineProperty(exports, "__esModule", { value: true });
/*!
 * devextreme-angular
 * Version: 19.1.11
 * Build date: Fri May 15 2020
 *
 * Copyright (c) 2012 - 2020 Developer Express Inc. ALL RIGHTS RESERVED
 *
 * This software may be modified and distributed under the terms
 * of the MIT license. See the LICENSE file in the root of the project for details.
 *
 * https://github.com/DevExpress/devextreme-angular
 */
__export(require("./core/component"));
__export(require("./core/events-strategy"));
__export(require("./core/integration"));
__export(require("./core/iterable-differ-helper"));
__export(require("./core/nested-option"));
__export(require("./core/template-host"));
__export(require("./core/template"));
__export(require("./core/transfer-state"));
__export(require("./core/utils"));
__export(require("./core/watcher-helper"));
__export(require("./ui/accordion"));
__export(require("./ui/action-sheet"));
__export(require("./ui/all"));
__export(require("./ui/autocomplete"));
__export(require("./ui/bar-gauge"));
__export(require("./ui/box"));
__export(require("./ui/bullet"));
__export(require("./ui/button-group"));
__export(require("./ui/button"));
__export(require("./ui/calendar"));
__export(require("./ui/chart"));
__export(require("./ui/check-box"));
__export(require("./ui/circular-gauge"));
__export(require("./ui/color-box"));
__export(require("./ui/context-menu"));
__export(require("./ui/data-grid"));
__export(require("./ui/date-box"));
__export(require("./ui/defer-rendering"));
__export(require("./ui/drawer"));
__export(require("./ui/drop-down-box"));
__export(require("./ui/drop-down-button"));
__export(require("./ui/file-uploader"));
__export(require("./ui/filter-builder"));
__export(require("./ui/form"));
__export(require("./ui/funnel"));
__export(require("./ui/gallery"));
__export(require("./ui/html-editor"));
__export(require("./ui/linear-gauge"));
__export(require("./ui/list"));
__export(require("./ui/load-indicator"));
__export(require("./ui/load-panel"));
__export(require("./ui/lookup"));
__export(require("./ui/map"));
__export(require("./ui/menu"));
__export(require("./ui/multi-view"));
__export(require("./ui/nav-bar"));
__export(require("./ui/number-box"));
__export(require("./ui/pie-chart"));
__export(require("./ui/pivot-grid-field-chooser"));
__export(require("./ui/pivot-grid"));
__export(require("./ui/polar-chart"));
__export(require("./ui/popover"));
__export(require("./ui/popup"));
__export(require("./ui/progress-bar"));
__export(require("./ui/radio-group"));
__export(require("./ui/range-selector"));
__export(require("./ui/range-slider"));
__export(require("./ui/recurrence-editor"));
__export(require("./ui/resizable"));
__export(require("./ui/responsive-box"));
__export(require("./ui/sankey"));
__export(require("./ui/scheduler"));
__export(require("./ui/scroll-view"));
__export(require("./ui/select-box"));
__export(require("./ui/slide-out-view"));
__export(require("./ui/slide-out"));
__export(require("./ui/slider"));
__export(require("./ui/sparkline"));
__export(require("./ui/speed-dial-action"));
__export(require("./ui/switch"));
__export(require("./ui/tab-panel"));
__export(require("./ui/tabs"));
__export(require("./ui/tag-box"));
__export(require("./ui/text-area"));
__export(require("./ui/text-box"));
__export(require("./ui/tile-view"));
__export(require("./ui/toast"));
__export(require("./ui/toolbar"));
__export(require("./ui/tooltip"));
__export(require("./ui/tree-list"));
__export(require("./ui/tree-map"));
__export(require("./ui/tree-view"));
__export(require("./ui/validation-group"));
__export(require("./ui/validation-summary"));
__export(require("./ui/validator"));
__export(require("./ui/vector-map"));

exports.DxiValidationRuleComponent = ɵngcc0.DxiValidationRuleComponent;
exports.DxiTabComponent = ɵngcc1.DxiTabComponent;
exports.DxiLocationComponent = ɵngcc2.DxiLocationComponent;
exports.DxiItemComponent = ɵngcc3.DxiItemComponent;
exports.DxiItemModule = ɵngcc3.DxiItemModule;
exports.DxiButtonComponent = ɵngcc4.DxiButtonComponent;
exports.DxoOptionsComponent = ɵngcc5.DxoOptionsComponent;
exports.DxiButtonModule = ɵngcc4.DxiButtonModule;
exports.DxoOptionsModule = ɵngcc5.DxoOptionsModule;
exports.DxoAnimationComponent = ɵngcc6.DxoAnimationComponent;
exports.DxoExportComponent = ɵngcc7.DxoExportComponent;
exports.DxoGeometryComponent = ɵngcc8.DxoGeometryComponent;
exports.DxoLabelComponent = ɵngcc9.DxoLabelComponent;
exports.DxoFontComponent = ɵngcc10.DxoFontComponent;
exports.DxoFormatComponent = ɵngcc11.DxoFormatComponent;
exports.DxoLegendComponent = ɵngcc12.DxoLegendComponent;
exports.DxoBorderComponent = ɵngcc13.DxoBorderComponent;
exports.DxoItemTextFormatComponent = ɵngcc14.DxoItemTextFormatComponent;
exports.DxoMarginComponent = ɵngcc15.DxoMarginComponent;
exports.DxoTitleComponent = ɵngcc16.DxoTitleComponent;
exports.DxoSubtitleComponent = ɵngcc17.DxoSubtitleComponent;
exports.DxoLoadingIndicatorComponent = ɵngcc18.DxoLoadingIndicatorComponent;
exports.DxoSizeComponent = ɵngcc19.DxoSizeComponent;
exports.DxoTooltipComponent = ɵngcc20.DxoTooltipComponent;
exports.DxoShadowComponent = ɵngcc21.DxoShadowComponent;
exports.DxoAnimationModule = ɵngcc6.DxoAnimationModule;
exports.DxoExportModule = ɵngcc7.DxoExportModule;
exports.DxoGeometryModule = ɵngcc8.DxoGeometryModule;
exports.DxoLabelModule = ɵngcc9.DxoLabelModule;
exports.DxoFontModule = ɵngcc10.DxoFontModule;
exports.DxoFormatModule = ɵngcc11.DxoFormatModule;
exports.DxoLegendModule = ɵngcc12.DxoLegendModule;
exports.DxoBorderModule = ɵngcc13.DxoBorderModule;
exports.DxoItemTextFormatModule = ɵngcc14.DxoItemTextFormatModule;
exports.DxoMarginModule = ɵngcc15.DxoMarginModule;
exports.DxoTitleModule = ɵngcc16.DxoTitleModule;
exports.DxoSubtitleModule = ɵngcc17.DxoSubtitleModule;
exports.DxoLoadingIndicatorModule = ɵngcc18.DxoLoadingIndicatorModule;
exports.DxoSizeModule = ɵngcc19.DxoSizeModule;
exports.DxoTooltipModule = ɵngcc20.DxoTooltipModule;
exports.DxoShadowModule = ɵngcc21.DxoShadowModule;
exports.DxoBoxComponent = ɵngcc22.DxoBoxComponent;
exports.DxoBoxModule = ɵngcc22.DxoBoxModule;
exports.DxoAdaptiveLayoutComponent = ɵngcc23.DxoAdaptiveLayoutComponent;
exports.DxiAnnotationComponent = ɵngcc24.DxiAnnotationComponent;
exports.DxoImageComponent = ɵngcc25.DxoImageComponent;
exports.DxiBreakComponent = ɵngcc26.DxiBreakComponent;
exports.DxiConstantLineComponent = ɵngcc27.DxiConstantLineComponent;
exports.DxiStripComponent = ɵngcc28.DxiStripComponent;
exports.DxoArgumentAxisComponent = ɵngcc29.DxoArgumentAxisComponent;
exports.DxoAggregationIntervalComponent = ɵngcc30.DxoAggregationIntervalComponent;
exports.DxoBreakStyleComponent = ɵngcc31.DxoBreakStyleComponent;
exports.DxoConstantLineStyleComponent = ɵngcc32.DxoConstantLineStyleComponent;
exports.DxoGridComponent = ɵngcc33.DxoGridComponent;
exports.DxoMinorGridComponent = ɵngcc34.DxoMinorGridComponent;
exports.DxoMinorTickComponent = ɵngcc35.DxoMinorTickComponent;
exports.DxoMinorTickIntervalComponent = ɵngcc36.DxoMinorTickIntervalComponent;
exports.DxoMinVisualRangeLengthComponent = ɵngcc37.DxoMinVisualRangeLengthComponent;
exports.DxoStripStyleComponent = ɵngcc38.DxoStripStyleComponent;
exports.DxoTickComponent = ɵngcc39.DxoTickComponent;
exports.DxoTickIntervalComponent = ɵngcc40.DxoTickIntervalComponent;
exports.DxoCommonAnnotationSettingsComponent = ɵngcc41.DxoCommonAnnotationSettingsComponent;
exports.DxoCommonAxisSettingsComponent = ɵngcc42.DxoCommonAxisSettingsComponent;
exports.DxoCommonPaneSettingsComponent = ɵngcc43.DxoCommonPaneSettingsComponent;
exports.DxoCommonSeriesSettingsComponent = ɵngcc44.DxoCommonSeriesSettingsComponent;
exports.DxoAggregationComponent = ɵngcc45.DxoAggregationComponent;
exports.DxoAreaComponent = ɵngcc46.DxoAreaComponent;
exports.DxoHoverStyleComponent = ɵngcc47.DxoHoverStyleComponent;
exports.DxoHatchingComponent = ɵngcc48.DxoHatchingComponent;
exports.DxoConnectorComponent = ɵngcc49.DxoConnectorComponent;
exports.DxoPointComponent = ɵngcc50.DxoPointComponent;
exports.DxoHeightComponent = ɵngcc51.DxoHeightComponent;
exports.DxoUrlComponent = ɵngcc52.DxoUrlComponent;
exports.DxoWidthComponent = ɵngcc53.DxoWidthComponent;
exports.DxoSelectionStyleComponent = ɵngcc54.DxoSelectionStyleComponent;
exports.DxoReductionComponent = ɵngcc55.DxoReductionComponent;
exports.DxoValueErrorBarComponent = ɵngcc56.DxoValueErrorBarComponent;
exports.DxoBarComponent = ɵngcc57.DxoBarComponent;
exports.DxoBubbleComponent = ɵngcc58.DxoBubbleComponent;
exports.DxoCandlestickComponent = ɵngcc59.DxoCandlestickComponent;
exports.DxoFullstackedareaComponent = ɵngcc60.DxoFullstackedareaComponent;
exports.DxoFullstackedbarComponent = ɵngcc61.DxoFullstackedbarComponent;
exports.DxoFullstackedlineComponent = ɵngcc62.DxoFullstackedlineComponent;
exports.DxoFullstackedsplineComponent = ɵngcc63.DxoFullstackedsplineComponent;
exports.DxoFullstackedsplineareaComponent = ɵngcc64.DxoFullstackedsplineareaComponent;
exports.DxoArgumentFormatComponent = ɵngcc65.DxoArgumentFormatComponent;
exports.DxoLineComponent = ɵngcc66.DxoLineComponent;
exports.DxoRangeareaComponent = ɵngcc67.DxoRangeareaComponent;
exports.DxoRangebarComponent = ɵngcc68.DxoRangebarComponent;
exports.DxoScatterComponent = ɵngcc69.DxoScatterComponent;
exports.DxoSplineComponent = ɵngcc70.DxoSplineComponent;
exports.DxoSplineareaComponent = ɵngcc71.DxoSplineareaComponent;
exports.DxoStackedareaComponent = ɵngcc72.DxoStackedareaComponent;
exports.DxoStackedbarComponent = ɵngcc73.DxoStackedbarComponent;
exports.DxoStackedlineComponent = ɵngcc74.DxoStackedlineComponent;
exports.DxoStackedsplineComponent = ɵngcc75.DxoStackedsplineComponent;
exports.DxoStackedsplineareaComponent = ɵngcc76.DxoStackedsplineareaComponent;
exports.DxoStepareaComponent = ɵngcc77.DxoStepareaComponent;
exports.DxoSteplineComponent = ɵngcc78.DxoSteplineComponent;
exports.DxoStockComponent = ɵngcc79.DxoStockComponent;
exports.DxoCrosshairComponent = ɵngcc80.DxoCrosshairComponent;
exports.DxoHorizontalLineComponent = ɵngcc81.DxoHorizontalLineComponent;
exports.DxoVerticalLineComponent = ɵngcc82.DxoVerticalLineComponent;
exports.DxoDataPrepareSettingsComponent = ɵngcc83.DxoDataPrepareSettingsComponent;
exports.DxiPaneComponent = ɵngcc84.DxiPaneComponent;
exports.DxoScrollBarComponent = ɵngcc85.DxoScrollBarComponent;
exports.DxiSeriesComponent = ɵngcc86.DxiSeriesComponent;
exports.DxoSeriesTemplateComponent = ɵngcc87.DxoSeriesTemplateComponent;
exports.DxiValueAxisComponent = ɵngcc88.DxiValueAxisComponent;
exports.DxoZoomAndPanComponent = ɵngcc89.DxoZoomAndPanComponent;
exports.DxoDragBoxStyleComponent = ɵngcc90.DxoDragBoxStyleComponent;
exports.DxoAdaptiveLayoutModule = ɵngcc23.DxoAdaptiveLayoutModule;
exports.DxiAnnotationModule = ɵngcc24.DxiAnnotationModule;
exports.DxoImageModule = ɵngcc25.DxoImageModule;
exports.DxoArgumentAxisModule = ɵngcc29.DxoArgumentAxisModule;
exports.DxoAggregationIntervalModule = ɵngcc30.DxoAggregationIntervalModule;
exports.DxiBreakModule = ɵngcc26.DxiBreakModule;
exports.DxoBreakStyleModule = ɵngcc31.DxoBreakStyleModule;
exports.DxiConstantLineModule = ɵngcc27.DxiConstantLineModule;
exports.DxoConstantLineStyleModule = ɵngcc32.DxoConstantLineStyleModule;
exports.DxoGridModule = ɵngcc33.DxoGridModule;
exports.DxoMinorGridModule = ɵngcc34.DxoMinorGridModule;
exports.DxoMinorTickModule = ɵngcc35.DxoMinorTickModule;
exports.DxoMinorTickIntervalModule = ɵngcc36.DxoMinorTickIntervalModule;
exports.DxoMinVisualRangeLengthModule = ɵngcc37.DxoMinVisualRangeLengthModule;
exports.DxiStripModule = ɵngcc28.DxiStripModule;
exports.DxoStripStyleModule = ɵngcc38.DxoStripStyleModule;
exports.DxoTickModule = ɵngcc39.DxoTickModule;
exports.DxoTickIntervalModule = ɵngcc40.DxoTickIntervalModule;
exports.DxoCommonAnnotationSettingsModule = ɵngcc41.DxoCommonAnnotationSettingsModule;
exports.DxoCommonAxisSettingsModule = ɵngcc42.DxoCommonAxisSettingsModule;
exports.DxoCommonPaneSettingsModule = ɵngcc43.DxoCommonPaneSettingsModule;
exports.DxoCommonSeriesSettingsModule = ɵngcc44.DxoCommonSeriesSettingsModule;
exports.DxoAggregationModule = ɵngcc45.DxoAggregationModule;
exports.DxoAreaModule = ɵngcc46.DxoAreaModule;
exports.DxoHoverStyleModule = ɵngcc47.DxoHoverStyleModule;
exports.DxoHatchingModule = ɵngcc48.DxoHatchingModule;
exports.DxoConnectorModule = ɵngcc49.DxoConnectorModule;
exports.DxoPointModule = ɵngcc50.DxoPointModule;
exports.DxoHeightModule = ɵngcc51.DxoHeightModule;
exports.DxoUrlModule = ɵngcc52.DxoUrlModule;
exports.DxoWidthModule = ɵngcc53.DxoWidthModule;
exports.DxoSelectionStyleModule = ɵngcc54.DxoSelectionStyleModule;
exports.DxoReductionModule = ɵngcc55.DxoReductionModule;
exports.DxoValueErrorBarModule = ɵngcc56.DxoValueErrorBarModule;
exports.DxoBarModule = ɵngcc57.DxoBarModule;
exports.DxoBubbleModule = ɵngcc58.DxoBubbleModule;
exports.DxoCandlestickModule = ɵngcc59.DxoCandlestickModule;
exports.DxoFullstackedareaModule = ɵngcc60.DxoFullstackedareaModule;
exports.DxoFullstackedbarModule = ɵngcc61.DxoFullstackedbarModule;
exports.DxoFullstackedlineModule = ɵngcc62.DxoFullstackedlineModule;
exports.DxoFullstackedsplineModule = ɵngcc63.DxoFullstackedsplineModule;
exports.DxoFullstackedsplineareaModule = ɵngcc64.DxoFullstackedsplineareaModule;
exports.DxoArgumentFormatModule = ɵngcc65.DxoArgumentFormatModule;
exports.DxoLineModule = ɵngcc66.DxoLineModule;
exports.DxoRangeareaModule = ɵngcc67.DxoRangeareaModule;
exports.DxoRangebarModule = ɵngcc68.DxoRangebarModule;
exports.DxoScatterModule = ɵngcc69.DxoScatterModule;
exports.DxoSplineModule = ɵngcc70.DxoSplineModule;
exports.DxoSplineareaModule = ɵngcc71.DxoSplineareaModule;
exports.DxoStackedareaModule = ɵngcc72.DxoStackedareaModule;
exports.DxoStackedbarModule = ɵngcc73.DxoStackedbarModule;
exports.DxoStackedlineModule = ɵngcc74.DxoStackedlineModule;
exports.DxoStackedsplineModule = ɵngcc75.DxoStackedsplineModule;
exports.DxoStackedsplineareaModule = ɵngcc76.DxoStackedsplineareaModule;
exports.DxoStepareaModule = ɵngcc77.DxoStepareaModule;
exports.DxoSteplineModule = ɵngcc78.DxoSteplineModule;
exports.DxoStockModule = ɵngcc79.DxoStockModule;
exports.DxoCrosshairModule = ɵngcc80.DxoCrosshairModule;
exports.DxoHorizontalLineModule = ɵngcc81.DxoHorizontalLineModule;
exports.DxoVerticalLineModule = ɵngcc82.DxoVerticalLineModule;
exports.DxoDataPrepareSettingsModule = ɵngcc83.DxoDataPrepareSettingsModule;
exports.DxiPaneModule = ɵngcc84.DxiPaneModule;
exports.DxoScrollBarModule = ɵngcc85.DxoScrollBarModule;
exports.DxiSeriesModule = ɵngcc86.DxiSeriesModule;
exports.DxoSeriesTemplateModule = ɵngcc87.DxoSeriesTemplateModule;
exports.DxiValueAxisModule = ɵngcc88.DxiValueAxisModule;
exports.DxoZoomAndPanModule = ɵngcc89.DxoZoomAndPanModule;
exports.DxoDragBoxStyleModule = ɵngcc90.DxoDragBoxStyleModule;
exports.DxiRangeComponent = ɵngcc91.DxiRangeComponent;
exports.DxoRangeContainerComponent = ɵngcc92.DxoRangeContainerComponent;
exports.DxoScaleComponent = ɵngcc93.DxoScaleComponent;
exports.DxoSubvalueIndicatorComponent = ɵngcc94.DxoSubvalueIndicatorComponent;
exports.DxoTextComponent = ɵngcc95.DxoTextComponent;
exports.DxoValueIndicatorComponent = ɵngcc96.DxoValueIndicatorComponent;
exports.DxoRangeContainerModule = ɵngcc92.DxoRangeContainerModule;
exports.DxiRangeModule = ɵngcc91.DxiRangeModule;
exports.DxoScaleModule = ɵngcc93.DxoScaleModule;
exports.DxoSubvalueIndicatorModule = ɵngcc94.DxoSubvalueIndicatorModule;
exports.DxoTextModule = ɵngcc95.DxoTextModule;
exports.DxoValueIndicatorModule = ɵngcc96.DxoValueIndicatorModule;
exports.DxoHideComponent = ɵngcc97.DxoHideComponent;
exports.DxoShowComponent = ɵngcc98.DxoShowComponent;
exports.DxoPositionComponent = ɵngcc99.DxoPositionComponent;
exports.DxoAtComponent = ɵngcc100.DxoAtComponent;
exports.DxoBoundaryOffsetComponent = ɵngcc101.DxoBoundaryOffsetComponent;
exports.DxoCollisionComponent = ɵngcc102.DxoCollisionComponent;
exports.DxoMyComponent = ɵngcc103.DxoMyComponent;
exports.DxoOffsetComponent = ɵngcc104.DxoOffsetComponent;
exports.DxoShowEventComponent = ɵngcc105.DxoShowEventComponent;
exports.DxoShowSubmenuModeComponent = ɵngcc106.DxoShowSubmenuModeComponent;
exports.DxoDelayComponent = ɵngcc107.DxoDelayComponent;
exports.DxoHideModule = ɵngcc97.DxoHideModule;
exports.DxoShowModule = ɵngcc98.DxoShowModule;
exports.DxoPositionModule = ɵngcc99.DxoPositionModule;
exports.DxoAtModule = ɵngcc100.DxoAtModule;
exports.DxoBoundaryOffsetModule = ɵngcc101.DxoBoundaryOffsetModule;
exports.DxoCollisionModule = ɵngcc102.DxoCollisionModule;
exports.DxoMyModule = ɵngcc103.DxoMyModule;
exports.DxoOffsetModule = ɵngcc104.DxoOffsetModule;
exports.DxoShowEventModule = ɵngcc105.DxoShowEventModule;
exports.DxoShowSubmenuModeModule = ɵngcc106.DxoShowSubmenuModeModule;
exports.DxoDelayModule = ɵngcc107.DxoDelayModule;
exports.DxoColumnChooserComponent = ɵngcc108.DxoColumnChooserComponent;
exports.DxoColumnFixingComponent = ɵngcc109.DxoColumnFixingComponent;
exports.DxoTextsComponent = ɵngcc110.DxoTextsComponent;
exports.DxiColumnComponent = ɵngcc111.DxiColumnComponent;
exports.DxoHeaderFilterComponent = ɵngcc112.DxoHeaderFilterComponent;
exports.DxoLookupComponent = ɵngcc113.DxoLookupComponent;
exports.DxoFormItemComponent = ɵngcc114.DxoFormItemComponent;
exports.DxoEditingComponent = ɵngcc115.DxoEditingComponent;
exports.DxoFormComponent = ɵngcc116.DxoFormComponent;
exports.DxoColCountByScreenComponent = ɵngcc117.DxoColCountByScreenComponent;
exports.DxoTabPanelOptionsComponent = ɵngcc118.DxoTabPanelOptionsComponent;
exports.DxoButtonOptionsComponent = ɵngcc119.DxoButtonOptionsComponent;
exports.DxiToolbarItemComponent = ɵngcc120.DxiToolbarItemComponent;
exports.DxoPopupComponent = ɵngcc121.DxoPopupComponent;
exports.DxiCustomOperationComponent = ɵngcc122.DxiCustomOperationComponent;
exports.DxiFieldComponent = ɵngcc123.DxiFieldComponent;
exports.DxoFilterBuilderComponent = ɵngcc124.DxoFilterBuilderComponent;
exports.DxoFilterOperationDescriptionsComponent = ɵngcc125.DxoFilterOperationDescriptionsComponent;
exports.DxoGroupOperationDescriptionsComponent = ɵngcc126.DxoGroupOperationDescriptionsComponent;
exports.DxoFilterBuilderPopupComponent = ɵngcc127.DxoFilterBuilderPopupComponent;
exports.DxoFilterPanelComponent = ɵngcc128.DxoFilterPanelComponent;
exports.DxoFilterRowComponent = ɵngcc129.DxoFilterRowComponent;
exports.DxoOperationDescriptionsComponent = ɵngcc130.DxoOperationDescriptionsComponent;
exports.DxoGroupingComponent = ɵngcc131.DxoGroupingComponent;
exports.DxoGroupPanelComponent = ɵngcc132.DxoGroupPanelComponent;
exports.DxoKeyboardNavigationComponent = ɵngcc133.DxoKeyboardNavigationComponent;
exports.DxoLoadPanelComponent = ɵngcc134.DxoLoadPanelComponent;
exports.DxoMasterDetailComponent = ɵngcc135.DxoMasterDetailComponent;
exports.DxoPagerComponent = ɵngcc136.DxoPagerComponent;
exports.DxoPagingComponent = ɵngcc137.DxoPagingComponent;
exports.DxoRemoteOperationsComponent = ɵngcc138.DxoRemoteOperationsComponent;
exports.DxoScrollingComponent = ɵngcc139.DxoScrollingComponent;
exports.DxoSearchPanelComponent = ɵngcc140.DxoSearchPanelComponent;
exports.DxoSelectionComponent = ɵngcc141.DxoSelectionComponent;
exports.DxiSortByGroupSummaryInfoComponent = ɵngcc142.DxiSortByGroupSummaryInfoComponent;
exports.DxoSortingComponent = ɵngcc143.DxoSortingComponent;
exports.DxoStateStoringComponent = ɵngcc144.DxoStateStoringComponent;
exports.DxiGroupItemComponent = ɵngcc145.DxiGroupItemComponent;
exports.DxiTotalItemComponent = ɵngcc146.DxiTotalItemComponent;
exports.DxoSummaryComponent = ɵngcc147.DxoSummaryComponent;
exports.DxoValueFormatComponent = ɵngcc148.DxoValueFormatComponent;
exports.DxoColumnChooserModule = ɵngcc108.DxoColumnChooserModule;
exports.DxoColumnFixingModule = ɵngcc109.DxoColumnFixingModule;
exports.DxoTextsModule = ɵngcc110.DxoTextsModule;
exports.DxiColumnModule = ɵngcc111.DxiColumnModule;
exports.DxoHeaderFilterModule = ɵngcc112.DxoHeaderFilterModule;
exports.DxoLookupModule = ɵngcc113.DxoLookupModule;
exports.DxoFormItemModule = ɵngcc114.DxoFormItemModule;
exports.DxiValidationRuleModule = ɵngcc0.DxiValidationRuleModule;
exports.DxoEditingModule = ɵngcc115.DxoEditingModule;
exports.DxoFormModule = ɵngcc116.DxoFormModule;
exports.DxoColCountByScreenModule = ɵngcc117.DxoColCountByScreenModule;
exports.DxoTabPanelOptionsModule = ɵngcc118.DxoTabPanelOptionsModule;
exports.DxiTabModule = ɵngcc1.DxiTabModule;
exports.DxoButtonOptionsModule = ɵngcc119.DxoButtonOptionsModule;
exports.DxoPopupModule = ɵngcc121.DxoPopupModule;
exports.DxiToolbarItemModule = ɵngcc120.DxiToolbarItemModule;
exports.DxoFilterBuilderModule = ɵngcc124.DxoFilterBuilderModule;
exports.DxiCustomOperationModule = ɵngcc122.DxiCustomOperationModule;
exports.DxiFieldModule = ɵngcc123.DxiFieldModule;
exports.DxoFilterOperationDescriptionsModule = ɵngcc125.DxoFilterOperationDescriptionsModule;
exports.DxoGroupOperationDescriptionsModule = ɵngcc126.DxoGroupOperationDescriptionsModule;
exports.DxoFilterBuilderPopupModule = ɵngcc127.DxoFilterBuilderPopupModule;
exports.DxoFilterPanelModule = ɵngcc128.DxoFilterPanelModule;
exports.DxoFilterRowModule = ɵngcc129.DxoFilterRowModule;
exports.DxoOperationDescriptionsModule = ɵngcc130.DxoOperationDescriptionsModule;
exports.DxoGroupingModule = ɵngcc131.DxoGroupingModule;
exports.DxoGroupPanelModule = ɵngcc132.DxoGroupPanelModule;
exports.DxoKeyboardNavigationModule = ɵngcc133.DxoKeyboardNavigationModule;
exports.DxoLoadPanelModule = ɵngcc134.DxoLoadPanelModule;
exports.DxoMasterDetailModule = ɵngcc135.DxoMasterDetailModule;
exports.DxoPagerModule = ɵngcc136.DxoPagerModule;
exports.DxoPagingModule = ɵngcc137.DxoPagingModule;
exports.DxoRemoteOperationsModule = ɵngcc138.DxoRemoteOperationsModule;
exports.DxoScrollingModule = ɵngcc139.DxoScrollingModule;
exports.DxoSearchPanelModule = ɵngcc140.DxoSearchPanelModule;
exports.DxoSelectionModule = ɵngcc141.DxoSelectionModule;
exports.DxiSortByGroupSummaryInfoModule = ɵngcc142.DxiSortByGroupSummaryInfoModule;
exports.DxoSortingModule = ɵngcc143.DxoSortingModule;
exports.DxoStateStoringModule = ɵngcc144.DxoStateStoringModule;
exports.DxoSummaryModule = ɵngcc147.DxoSummaryModule;
exports.DxiGroupItemModule = ɵngcc145.DxiGroupItemModule;
exports.DxoValueFormatModule = ɵngcc148.DxoValueFormatModule;
exports.DxiTotalItemModule = ɵngcc146.DxiTotalItemModule;
exports.DxoCalendarOptionsComponent = ɵngcc149.DxoCalendarOptionsComponent;
exports.DxoDisplayFormatComponent = ɵngcc150.DxoDisplayFormatComponent;
exports.DxoCalendarOptionsModule = ɵngcc149.DxoCalendarOptionsModule;
exports.DxoDisplayFormatModule = ɵngcc150.DxoDisplayFormatModule;
exports.DxoDropDownOptionsComponent = ɵngcc151.DxoDropDownOptionsComponent;
exports.DxoDropDownOptionsModule = ɵngcc151.DxoDropDownOptionsModule;
exports.DxoItemComponent = ɵngcc152.DxoItemComponent;
exports.DxoItemModule = ɵngcc152.DxoItemModule;
exports.DxoMediaResizingComponent = ɵngcc153.DxoMediaResizingComponent;
exports.DxiMentionComponent = ɵngcc154.DxiMentionComponent;
exports.DxoToolbarComponent = ɵngcc155.DxoToolbarComponent;
exports.DxoVariablesComponent = ɵngcc156.DxoVariablesComponent;
exports.DxoMediaResizingModule = ɵngcc153.DxoMediaResizingModule;
exports.DxiMentionModule = ɵngcc154.DxiMentionModule;
exports.DxoToolbarModule = ɵngcc155.DxoToolbarModule;
exports.DxoVariablesModule = ɵngcc156.DxoVariablesModule;
exports.DxiMenuItemComponent = ɵngcc157.DxiMenuItemComponent;
exports.DxoSearchEditorOptionsComponent = ɵngcc158.DxoSearchEditorOptionsComponent;
exports.DxiMenuItemModule = ɵngcc157.DxiMenuItemModule;
exports.DxoSearchEditorOptionsModule = ɵngcc158.DxoSearchEditorOptionsModule;
exports.DxiCenterComponent = ɵngcc159.DxiCenterComponent;
exports.DxoKeyComponent = ɵngcc160.DxoKeyComponent;
exports.DxiMarkerComponent = ɵngcc161.DxiMarkerComponent;
exports.DxiRouteComponent = ɵngcc162.DxiRouteComponent;
exports.DxiCenterModule = ɵngcc159.DxiCenterModule;
exports.DxoKeyModule = ɵngcc160.DxoKeyModule;
exports.DxiMarkerModule = ɵngcc161.DxiMarkerModule;
exports.DxiLocationModule = ɵngcc2.DxiLocationModule;
exports.DxiRouteModule = ɵngcc162.DxiRouteModule;
exports.DxoShowFirstSubmenuModeComponent = ɵngcc163.DxoShowFirstSubmenuModeComponent;
exports.DxoShowFirstSubmenuModeModule = ɵngcc163.DxoShowFirstSubmenuModeModule;
exports.DxoSmallValuesGroupingComponent = ɵngcc164.DxoSmallValuesGroupingComponent;
exports.DxoSmallValuesGroupingModule = ɵngcc164.DxoSmallValuesGroupingModule;
exports.DxoDataSourceComponent = ɵngcc165.DxoDataSourceComponent;
exports.DxoStoreComponent = ɵngcc166.DxoStoreComponent;
exports.DxoFieldChooserComponent = ɵngcc167.DxoFieldChooserComponent;
exports.DxoFieldPanelComponent = ɵngcc168.DxoFieldPanelComponent;
exports.DxoDataSourceModule = ɵngcc165.DxoDataSourceModule;
exports.DxoStoreModule = ɵngcc166.DxoStoreModule;
exports.DxoFieldChooserModule = ɵngcc167.DxoFieldChooserModule;
exports.DxoFieldPanelModule = ɵngcc168.DxoFieldPanelModule;
exports.DxoValueAxisComponent = ɵngcc169.DxoValueAxisComponent;
exports.DxoValueAxisModule = ɵngcc169.DxoValueAxisModule;
exports.DxoHideEventComponent = ɵngcc170.DxoHideEventComponent;
exports.DxoHideEventModule = ɵngcc170.DxoHideEventModule;
exports.DxoBackgroundComponent = ɵngcc171.DxoBackgroundComponent;
exports.DxoBehaviorComponent = ɵngcc172.DxoBehaviorComponent;
exports.DxoChartComponent = ɵngcc173.DxoChartComponent;
exports.DxoIndentComponent = ɵngcc174.DxoIndentComponent;
exports.DxoMarkerComponent = ɵngcc175.DxoMarkerComponent;
exports.DxoMaxRangeComponent = ɵngcc176.DxoMaxRangeComponent;
exports.DxoMinRangeComponent = ɵngcc177.DxoMinRangeComponent;
exports.DxoShutterComponent = ɵngcc178.DxoShutterComponent;
exports.DxoSliderHandleComponent = ɵngcc179.DxoSliderHandleComponent;
exports.DxoSliderMarkerComponent = ɵngcc180.DxoSliderMarkerComponent;
exports.DxoBackgroundModule = ɵngcc171.DxoBackgroundModule;
exports.DxoBehaviorModule = ɵngcc172.DxoBehaviorModule;
exports.DxoChartModule = ɵngcc173.DxoChartModule;
exports.DxoIndentModule = ɵngcc174.DxoIndentModule;
exports.DxoMarkerModule = ɵngcc175.DxoMarkerModule;
exports.DxoMaxRangeModule = ɵngcc176.DxoMaxRangeModule;
exports.DxoMinRangeModule = ɵngcc177.DxoMinRangeModule;
exports.DxoShutterModule = ɵngcc178.DxoShutterModule;
exports.DxoSliderHandleModule = ɵngcc179.DxoSliderHandleModule;
exports.DxoSliderMarkerModule = ɵngcc180.DxoSliderMarkerModule;
exports.DxiColComponent = ɵngcc181.DxiColComponent;
exports.DxiRowComponent = ɵngcc182.DxiRowComponent;
exports.DxiColModule = ɵngcc181.DxiColModule;
exports.DxiRowModule = ɵngcc182.DxiRowModule;
exports.DxoLinkComponent = ɵngcc183.DxoLinkComponent;
exports.DxoNodeComponent = ɵngcc184.DxoNodeComponent;
exports.DxoLinkModule = ɵngcc183.DxoLinkModule;
exports.DxoNodeModule = ɵngcc184.DxoNodeModule;
exports.DxiResourceComponent = ɵngcc185.DxiResourceComponent;
exports.DxiViewComponent = ɵngcc186.DxiViewComponent;
exports.DxiResourceModule = ɵngcc185.DxiResourceModule;
exports.DxiViewModule = ɵngcc186.DxiViewModule;
exports.DxoColorizerComponent = ɵngcc187.DxoColorizerComponent;
exports.DxoGroupComponent = ɵngcc188.DxoGroupComponent;
exports.DxoTileComponent = ɵngcc189.DxoTileComponent;
exports.DxoColorizerModule = ɵngcc187.DxoColorizerModule;
exports.DxoGroupModule = ɵngcc188.DxoGroupModule;
exports.DxoTileModule = ɵngcc189.DxoTileModule;
exports.DxoAdapterComponent = ɵngcc190.DxoAdapterComponent;
exports.DxoAdapterModule = ɵngcc190.DxoAdapterModule;
exports.DxoControlBarComponent = ɵngcc191.DxoControlBarComponent;
exports.DxiLayerComponent = ɵngcc192.DxiLayerComponent;
exports.DxiLegendComponent = ɵngcc193.DxiLegendComponent;
exports.DxoSourceComponent = ɵngcc194.DxoSourceComponent;
exports.DxoControlBarModule = ɵngcc191.DxoControlBarModule;
exports.DxiLayerModule = ɵngcc192.DxiLayerModule;
exports.DxiLegendModule = ɵngcc193.DxiLegendModule;
exports.DxoSourceModule = ɵngcc194.DxoSourceModule;
