// devextreme-angular/__ivy_ngcc__/ui/nested/aggregation.js
"use strict";
var ɵngcc0 = require('@angular/core');
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
/*!
 * devextreme-angular
 * Version: 19.1.11
 * Build date: Fri May 15 2020
 *
 * Copyright (c) 2012 - 2020 Developer Express Inc. ALL RIGHTS RESERVED
 *
 * This software may be modified and distributed under the terms
 * of the MIT license. See the LICENSE file in the root of the project for details.
 *
 * https://github.com/DevExpress/devextreme-angular
 */
Object.defineProperty(exports, "__esModule", { value: true });
var core_1 = require("@angular/core");
var nested_option_1 = require("../../core/nested-option");
var nested_option_2 = require("../../core/nested-option");
var DxoAggregationComponent = (function (_super) {
    __extends(DxoAggregationComponent, _super);
    function DxoAggregationComponent(parentOptionHost, optionHost) {
        var _this = _super.call(this) || this;
        parentOptionHost.setNestedOption(_this);
        optionHost.setHost(_this, _this._fullOptionPath.bind(_this));
        return _this;
    }
    Object.defineProperty(DxoAggregationComponent.prototype, "calculate", {
        get: function () {
            return this._getOption('calculate');
        },
        set: function (value) {
            this._setOption('calculate', value);
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(DxoAggregationComponent.prototype, "enabled", {
        get: function () {
            return this._getOption('enabled');
        },
        set: function (value) {
            this._setOption('enabled', value);
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(DxoAggregationComponent.prototype, "method", {
        get: function () {
            return this._getOption('method');
        },
        set: function (value) {
            this._setOption('method', value);
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(DxoAggregationComponent.prototype, "_optionPath", {
        get: function () {
            return 'aggregation';
        },
        enumerable: true,
        configurable: true
    });
    /** @nocollapse */
    DxoAggregationComponent.ctorParameters = function () { return [
        { type: nested_option_1.NestedOptionHost, decorators: [{ type: core_1.SkipSelf }, { type: core_1.Host },] },
        { type: nested_option_1.NestedOptionHost, decorators: [{ type: core_1.Host },] },
    ]; };
    DxoAggregationComponent.propDecorators = {
        "calculate": [{ type: core_1.Input },],
        "enabled": [{ type: core_1.Input },],
        "method": [{ type: core_1.Input },],
    };
DxoAggregationComponent.ɵfac = function DxoAggregationComponent_Factory(t) { return new (t || DxoAggregationComponent)(ɵngcc0.ɵɵdirectiveInject(nested_option_1.NestedOptionHost, 5), ɵngcc0.ɵɵdirectiveInject(nested_option_1.NestedOptionHost, 1)); };
DxoAggregationComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: DxoAggregationComponent, selectors: [["dxo-aggregation"]], inputs: { calculate: "calculate", enabled: "enabled", method: "method" }, features: [ɵngcc0.ɵɵProvidersFeature([nested_option_1.NestedOptionHost]), ɵngcc0.ɵɵInheritDefinitionFeature], decls: 0, vars: 0, template: function DxoAggregationComponent_Template(rf, ctx) { }, styles: [""] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(DxoAggregationComponent, [{
        type: core_1.Component,
        args: [{
                selector: 'dxo-aggregation',
                template: '',
                styles: [''],
                providers: [nested_option_1.NestedOptionHost]
            }]
    }], function () { return [{ type: nested_option_1.NestedOptionHost, decorators: [{
                type: core_1.SkipSelf
            }, {
                type: core_1.Host
            }] }, { type: nested_option_1.NestedOptionHost, decorators: [{
                type: core_1.Host
            }] }]; }, { calculate: [{
            type: core_1.Input
        }], enabled: [{
            type: core_1.Input
        }], method: [{
            type: core_1.Input
        }] }); })();
    return DxoAggregationComponent;
}(nested_option_2.NestedOption));
exports.DxoAggregationComponent = DxoAggregationComponent;
var DxoAggregationModule = (function () {
    function DxoAggregationModule() {
    }
DxoAggregationModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: DxoAggregationModule });
DxoAggregationModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function DxoAggregationModule_Factory(t) { return new (t || DxoAggregationModule)(); } });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(DxoAggregationModule, { declarations: [DxoAggregationComponent], exports: [DxoAggregationComponent] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(DxoAggregationModule, [{
        type: core_1.NgModule,
        args: [{
                declarations: [
                    DxoAggregationComponent
                ],
                exports: [
                    DxoAggregationComponent
                ]
            }]
    }], function () { return []; }, null); })();
    return DxoAggregationModule;
}());
exports.DxoAggregationModule = DxoAggregationModule;
