// devextreme-angular/__ivy_ngcc__/ui/nested/filter-builder.js
"use strict";
var ɵngcc0 = require('@angular/core');
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
/*!
 * devextreme-angular
 * Version: 19.1.11
 * Build date: Fri May 15 2020
 *
 * Copyright (c) 2012 - 2020 Developer Express Inc. ALL RIGHTS RESERVED
 *
 * This software may be modified and distributed under the terms
 * of the MIT license. See the LICENSE file in the root of the project for details.
 *
 * https://github.com/DevExpress/devextreme-angular
 */
Object.defineProperty(exports, "__esModule", { value: true });
var core_1 = require("@angular/core");
var nested_option_1 = require("../../core/nested-option");
var filter_builder_options_1 = require("./base/filter-builder-options");
var custom_operation_dxi_1 = require("./custom-operation-dxi");
var field_dxi_1 = require("./field-dxi");
var DxoFilterBuilderComponent = (function (_super) {
    __extends(DxoFilterBuilderComponent, _super);
    function DxoFilterBuilderComponent(parentOptionHost, optionHost) {
        var _this = _super.call(this) || this;
        _this._createEventEmitters([
            { emit: 'valueChange' }
        ]);
        parentOptionHost.setNestedOption(_this);
        optionHost.setHost(_this, _this._fullOptionPath.bind(_this));
        return _this;
    }
    Object.defineProperty(DxoFilterBuilderComponent.prototype, "_optionPath", {
        get: function () {
            return 'filterBuilder';
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(DxoFilterBuilderComponent.prototype, "customOperationsChildren", {
        get: function () {
            return this._getOption('customOperations');
        },
        set: function (value) {
            this.setChildren('customOperations', value);
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(DxoFilterBuilderComponent.prototype, "fieldsChildren", {
        get: function () {
            return this._getOption('fields');
        },
        set: function (value) {
            this.setChildren('fields', value);
        },
        enumerable: true,
        configurable: true
    });
    /** @nocollapse */
    DxoFilterBuilderComponent.ctorParameters = function () { return [
        { type: nested_option_1.NestedOptionHost, decorators: [{ type: core_1.SkipSelf }, { type: core_1.Host },] },
        { type: nested_option_1.NestedOptionHost, decorators: [{ type: core_1.Host },] },
    ]; };
    DxoFilterBuilderComponent.propDecorators = {
        "valueChange": [{ type: core_1.Output },],
        "customOperationsChildren": [{ type: core_1.ContentChildren, args: [core_1.forwardRef(function () { return custom_operation_dxi_1.DxiCustomOperationComponent; }),] },],
        "fieldsChildren": [{ type: core_1.ContentChildren, args: [core_1.forwardRef(function () { return field_dxi_1.DxiFieldComponent; }),] },],
    };
DxoFilterBuilderComponent.ɵfac = function DxoFilterBuilderComponent_Factory(t) { return new (t || DxoFilterBuilderComponent)(ɵngcc0.ɵɵdirectiveInject(nested_option_1.NestedOptionHost, 5), ɵngcc0.ɵɵdirectiveInject(nested_option_1.NestedOptionHost, 1)); };
DxoFilterBuilderComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: DxoFilterBuilderComponent, selectors: [["dxo-filter-builder"]], contentQueries: function DxoFilterBuilderComponent_ContentQueries(rf, ctx, dirIndex) { if (rf & 1) {
        ɵngcc0.ɵɵcontentQuery(dirIndex, custom_operation_dxi_1.DxiCustomOperationComponent, false);
        ɵngcc0.ɵɵcontentQuery(dirIndex, field_dxi_1.DxiFieldComponent, false);
    } if (rf & 2) {
        var _t;
        ɵngcc0.ɵɵqueryRefresh(_t = ɵngcc0.ɵɵloadQuery()) && (ctx.customOperationsChildren = _t);
        ɵngcc0.ɵɵqueryRefresh(_t = ɵngcc0.ɵɵloadQuery()) && (ctx.fieldsChildren = _t);
    } }, inputs: { accessKey: "accessKey", activeStateEnabled: "activeStateEnabled", allowHierarchicalFields: "allowHierarchicalFields", customOperations: "customOperations", disabled: "disabled", elementAttr: "elementAttr", fields: "fields", filterOperationDescriptions: "filterOperationDescriptions", focusStateEnabled: "focusStateEnabled", groupOperationDescriptions: "groupOperationDescriptions", groupOperations: "groupOperations", height: "height", hint: "hint", hoverStateEnabled: "hoverStateEnabled", maxGroupLevel: "maxGroupLevel", onContentReady: "onContentReady", onDisposing: "onDisposing", onEditorPrepared: "onEditorPrepared", onEditorPreparing: "onEditorPreparing", onInitialized: "onInitialized", onOptionChanged: "onOptionChanged", onValueChanged: "onValueChanged", rtlEnabled: "rtlEnabled", tabIndex: "tabIndex", value: "value", visible: "visible", width: "width" }, outputs: { valueChange: "valueChange" }, features: [ɵngcc0.ɵɵProvidersFeature([nested_option_1.NestedOptionHost]), ɵngcc0.ɵɵInheritDefinitionFeature], decls: 0, vars: 0, template: function DxoFilterBuilderComponent_Template(rf, ctx) { }, styles: [""] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(DxoFilterBuilderComponent, [{
        type: core_1.Component,
        args: [{
                selector: 'dxo-filter-builder',
                template: '',
                styles: [''],
                providers: [nested_option_1.NestedOptionHost],
                inputs: [
                    'accessKey',
                    'activeStateEnabled',
                    'allowHierarchicalFields',
                    'customOperations',
                    'disabled',
                    'elementAttr',
                    'fields',
                    'filterOperationDescriptions',
                    'focusStateEnabled',
                    'groupOperationDescriptions',
                    'groupOperations',
                    'height',
                    'hint',
                    'hoverStateEnabled',
                    'maxGroupLevel',
                    'onContentReady',
                    'onDisposing',
                    'onEditorPrepared',
                    'onEditorPreparing',
                    'onInitialized',
                    'onOptionChanged',
                    'onValueChanged',
                    'rtlEnabled',
                    'tabIndex',
                    'value',
                    'visible',
                    'width'
                ]
            }]
    }], function () { return [{ type: nested_option_1.NestedOptionHost, decorators: [{
                type: core_1.SkipSelf
            }, {
                type: core_1.Host
            }] }, { type: nested_option_1.NestedOptionHost, decorators: [{
                type: core_1.Host
            }] }]; }, { customOperationsChildren: [{
            type: core_1.ContentChildren,
            args: [core_1.forwardRef(function () { return custom_operation_dxi_1.DxiCustomOperationComponent; })]
        }], fieldsChildren: [{
            type: core_1.ContentChildren,
            args: [core_1.forwardRef(function () { return field_dxi_1.DxiFieldComponent; })]
        }], valueChange: [{
            type: core_1.Output
        }] }); })();
    return DxoFilterBuilderComponent;
}(filter_builder_options_1.DxoFilterBuilderOptions));
exports.DxoFilterBuilderComponent = DxoFilterBuilderComponent;
var DxoFilterBuilderModule = (function () {
    function DxoFilterBuilderModule() {
    }
DxoFilterBuilderModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: DxoFilterBuilderModule });
DxoFilterBuilderModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function DxoFilterBuilderModule_Factory(t) { return new (t || DxoFilterBuilderModule)(); } });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(DxoFilterBuilderModule, { declarations: [DxoFilterBuilderComponent], exports: [DxoFilterBuilderComponent] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(DxoFilterBuilderModule, [{
        type: core_1.NgModule,
        args: [{
                declarations: [
                    DxoFilterBuilderComponent
                ],
                exports: [
                    DxoFilterBuilderComponent
                ]
            }]
    }], function () { return []; }, null); })();
    return DxoFilterBuilderModule;
}());
exports.DxoFilterBuilderModule = DxoFilterBuilderModule;
